ecscollector/collector.go

```diff
@@ -0,0 +1,270 @@
+		labelKeys, nil)
+
+	cpuSystemDesc = prometheus.NewDesc(
+		"ecs_cpu_system",
+		"Total system CPU usage.",
+		labelKeys, nil)
+
+	cpuNumOnlineDesc = prometheus.NewDesc(
+		"ecs_cpu_num_online",
+		"Number of online CPUs.",
+		labelKeys, nil)
+
+	memUsageDesc = prometheus.NewDesc(
+		"ecs_mem_total",
+		"Total memory usage.",
+		labelKeys, nil)
+
+	memMaxUsageDesc = prometheus.NewDesc(
+		"ecs_mem_usage",
+		"Maximum memory usage.",
+		labelKeys, nil)
+
+	memLimitDesc = prometheus.NewDesc(
+		"ecs_mem_limit",
+		"Memory limit.",
+		labelKeys, nil)
+
+	networkRxBytesDesc = prometheus.NewDesc(
+		"ecs_network_rx_bytes",
+		"Network recieved in bytes.",
+		labelKeysWithNetwork, nil)
+
+	networkRxPacketsDesc = prometheus.NewDesc(
+		"ecs_network_rx_packets",
+		"Network packets recieved.",
+		labelKeysWithNetwork, nil)
+
+	networkRxDroppedDesc = prometheus.NewDesc(
+		"ecs_network_rx_dropped",
+		"Network packets dropped in recieving.",
+		labelKeysWithNetwork, nil)
+
+	networkRxErrorsDesc = prometheus.NewDesc(
+		"ecs_network_rx_errors",
+		"Network errors in recieving.",
+		labelKeysWithNetwork, nil)
+
+	networkTxBytesDesc = prometheus.NewDesc(
+		"ecs_network_tx_bytes",
+		"Network transmitted in bytes.",
+		labelKeysWithNetwork, nil)
+
+	networkTxPacketsDesc = prometheus.NewDesc(
+		"ecs_network_tx_packets",
+		"Network packets transmitted.",
+		labelKeysWithNetwork, nil)
+
+	networkTxDroppedDesc = prometheus.NewDesc(
+		"ecs_network_tx_dropped",
+		"Network packets dropped in transmit.",
+		labelKeysWithNetwork, nil)
+
+	networkTxErrorsDesc = prometheus.NewDesc(
+		"ecs_network_tx_errors",
+		"Network errors in transmit.",
+		labelKeysWithNetwork, nil)
+
+	networkRxRateDesc = prometheus.NewDesc(
+		"ecs_network_rx_rate",
+		"Network recieved rate per second.",
+		labelKeys, nil)
+
+	networkTxRateDesc = prometheus.NewDesc(
+		"ecs_network_tx_rate",
+		"Network transmitted rate per second.",
+		labelKeys, nil)
+)
+
+var labelKeys = []string{
+	"cluster",
+	"task_az",
+	"task_family",
+	"task_family_revision",
+	// TODO(jbd): Add service name as a label.
+	"container",
+}
+
+var labelKeysWithNetwork = append(labelKeys,
+	"network_interface",
+)
+
+// NewCollector returns a new Collector that queries ECS metadata server
+// for ECS task and container metrics.
+func NewCollector(client *ecsmetadata.Client) prometheus.Collector {
+	return &collector{client: client}
+}
+
+type collector struct {
+	client *ecsmetadata.Client
+}
+
+func (c *collector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- numProcsDesc
+	ch <- cpuTotalDesc
+	ch <- cpuUserDesc
+	ch <- cpuKernelDesc
+	ch <- cpuSystemDesc
+	ch <- cpuNumOnlineDesc
+	ch <- memUsageDesc
+	ch <- memMaxUsageDesc
+	ch <- memLimitDesc
+	ch <- networkRxBytesDesc
+	ch <- networkRxPacketsDesc
+	ch <- networkRxDroppedDesc
+	ch <- networkRxErrorsDesc
+	ch <- networkTxBytesDesc
+	ch <- networkTxPacketsDesc
+	ch <- networkTxDroppedDesc
+	ch <- networkTxErrorsDesc
+	ch <- networkRxRateDesc
```

Comment:
If we have the raw metrics, we can get the rate easily. We generally only expose the raw metrics.

Reply:
Removed them.